frontstage/templates/surveys/surveys-upload-failure.html

```diff
@@ -11,17 +11,23 @@
 
         <div class="panel panel--error u-mb-s" data-qa="error">
             <div class="panel__header">
-                <h1 class="panel__title venus">Something went wrong</h1>
+                 {% if ".xlsx format" in error_info.text %}
+                    <h1 class="panel__title venus">Error uploading - incorrect file type</h1>
+                 {% elif "50 characters" in error_info.text %}
+                    <h1 class="panel__title venus">Error uploading - file name too long</h1>
+                 {% elif "20MB" in error_info.text %}
+                    <h1 class="panel__title venus">Error uploading - file size too large</h1>
+                 {% else %}
+                    <h1 class="panel__title venus">Something went wrong</h1>
+                 {% endif %}
             </div>
             <div class="panel__body" data-qa="error-body">
 
-                <ul class="error-message">
-                    <li>{{ error_info.text }}</li>
-                </ul>
-
-                <p class="error-message">
-                    Please try uploading your spreadsheet again
-                </p>
+                {% if ".xlsx format" not in error_info.text and "50 characters" not in error_info.text and "20MB" not in error_info.text%}
+                    <p class="error-message">Please try uploading your spreadsheet again</p>
+                {% else %}
+                    <p class="error-message">{{ error_info.text }}</p>
```

Comment:
If the collection instrument falls over/gives out a 500 this is either going to be empty or a huge long exception string.
Should probably have a proper default error message

Reply:
Ii checked with Nick on the default error and that's what he said, if none of the above so: extension, name length or file size then it's something went wrong